src/libproc_macro/lib.rs

```diff
@@ -125,6 +125,10 @@ pub mod __internal {
         fn register_attr_proc_macro(&mut self,
                                     name: &str,
                                     expand: fn(TokenStream, TokenStream) -> TokenStream);
+
+        fn register_bang_proc_macro(&mut self,
```

Comment:
I prefer `Bang` inside the compiler to avoid overloading "function". Also, we might want to use "proc macro function" for the underlying function item annotated with `#[proc_macro*]`.